tests/lib/tools/nested-state

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash -e
+
+show_help() {
+    echo "usage: prepare-env"
```

Comment:
What is the environment that is being prepared or cleaned up here?

Since we have prepare/restore everywhere else, could we rename `cleanup-env` to restore and `prepare-env` to just `prepare`? CC @mvo5 @anonymouse64 for comment as I'm less familiar with nested than the rest of the team.

Reply:
makes sense to call them prepare/restore. Thanks for the suggestion